drivers/sip_svc/sip_svc_ll.c

```diff
@@ -0,0 +1,919 @@
+		if (trans_idx == SIP_SVC_ID_INVALID) {
+			LOG_ERR("Fail to allocate transaction id");
+			k_mutex_unlock(&ctrl->data_mutex);
+			return SIP_SVC_ID_INVALID;
+		}
+		trans_id = sip_svc_plat_format_trans_id(c_idx, trans_idx);
+
+		/* Assign the trans id of this request */
+		SIP_SVC_PROTO_HEADER_SET_TRANS_ID(request->header, trans_id);
+
+		/* Map trans id to client, callback, response data addr */
+		if (sip_svc_ll_id_map_insert_item(ctrl->trans_id_map,
+			trans_id,
+			(void *)cb,
+			(void *)(request->resp_data_addr & 0xFFFFFFFF),
+			(void *)((request->resp_data_addr >> 32) & 0xFFFFFFFF),
+			(void *)(uint64_t)request->resp_data_size,
+			request->priv_data,
+			(void *)(uint64_t)c_idx) != 0) {
+
+			LOG_ERR("Fail to insert transaction id to map");
+			sip_svc_ll_id_mgr_free(ctrl->clients[c_idx].trans_idx_pool,
+						trans_idx);
+			k_mutex_unlock(&ctrl->data_mutex);
+			return SIP_SVC_ID_INVALID;
+		}
+
+		/* Insert request to MSGQ */
+		if (k_mutex_lock(&ctrl->req_msgq_mutex, K_FOREVER) == 0) {
+
+			if (k_msgq_put(&ctrl->req_msgq, sip_svc_request,
+				K_NO_WAIT) != 0) {
+				LOG_ERR("Request msgq full");
+				sip_svc_ll_id_map_remove_item(ctrl->trans_id_map, trans_id);
+				sip_svc_ll_id_mgr_free(ctrl->clients[c_idx].trans_idx_pool,
+							trans_idx);
+				k_mutex_unlock(&ctrl->req_msgq_mutex);
+				k_mutex_unlock(&ctrl->data_mutex);
+				return SIP_SVC_ID_INVALID;
+			}
+			k_mutex_unlock(&ctrl->req_msgq_mutex);
+		}
+		ctrl->clients[c_idx].active_trans_cnt++;
+
+		/* Create and run the thread */
+		if (!ctrl->tid) {
+			ctrl->tid = k_thread_create(&ctrl->thread, ctrl->stack,
+					CONFIG_ARM_SIP_SVC_THREAD_STACK_SIZE,
+					sip_svc_ll_thread, ctrl, NULL, NULL,
+					CONFIG_ARM_SIP_SVC_THREAD_PRIORITY,
+					0, K_NO_WAIT);
+			if (!ctrl->tid) {
+				LOG_ERR("Fail to spawn sip_svp thread");
+				sip_svc_ll_id_map_remove_item(ctrl->trans_id_map, trans_id);
+				sip_svc_ll_id_mgr_free(ctrl->clients[c_idx].trans_idx_pool,
+							trans_idx);
+				k_mutex_unlock(&ctrl->data_mutex);
+				return SIP_SVC_ID_INVALID;
+			}
+		} else {
+			k_thread_resume(ctrl->tid);
+		}
+
+		k_mutex_unlock(&ctrl->data_mutex);
+	}
+
+	return trans_id;
+}
+
+void *sip_svc_ll_get_priv_data(struct sip_svc_controller *ctrl,
+			       uint32_t c_token)
+{
+	uint32_t c_idx;
+
+	if (!ctrl)
+		return NULL;
+
+	if (k_mutex_lock(&ctrl->data_mutex, K_FOREVER) == 0) {
+		c_idx = sip_svc_ll_get_c_idx(ctrl, c_token);
+
+		if (c_idx == SIP_SVC_ID_INVALID) {
+			k_mutex_unlock(&ctrl->data_mutex);
+			return NULL;
+		}
+
+		k_mutex_unlock(&ctrl->data_mutex);
+		return ctrl->clients[c_idx].priv_data;
+	}
+
+	return NULL;
+}
+
+void sip_svc_ll_print_info(struct sip_svc_controller *ctrl)
+{
+	uint32_t i;
+	static const char * const state_str_list[] = {"INVALID",
+							"IDLE",
+							"OPEN",
+							"ABORT"};
+	const char *state_str = "UNKNOWN";
+
+	if (!ctrl)
+		printk("Invalid sip_svc controller\n");
+
+	printk("---------------------------------------\n");
```

Comment:
Why do you use `printk` here, but LOG_XX in previous functions? Can we switch all `printk` to LOG_XX?